relayer/crates/starknet-chain-components/src/types/wallet.rs

```diff
@@ -3,9 +3,37 @@ use starknet::core::types::Felt;
 
 use crate::impls::types::address::StarknetAddress;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum SigningKey {
+    File { file: String },
+    Felt { felt: Felt },
+}
+
+impl TryFrom<SigningKey> for Felt {
+    type Error = String;
+
+    fn try_from(value: SigningKey) -> Result<Self, Self::Error> {
+        match value {
+            SigningKey::File { file } => Ok(std::fs::read_to_string(file)
```

Comment:
This is an unexpected side effect, some more with blocking I/O, which would happen every time the field is used, which is not good.

Reply:
Can you please take over this PR? Maybe I have to do the same way you do it for `CosmosChainConfig`

https://github.com/informalsystems/hermes-sdk/blob/27168d1ee8c004d87b39d769e80c686d79ed53c5/crates/cosmos/cosmos-chain-components/src/impls/types/config.rs#L21-L22

Reply:
No problem, get some rest. I think this is lower priority for now.